experiment/image-bumper/main.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	imageRegexp = regexp.MustCompile(`\b(gcr\.io)/([a-z][a-z0-9-]{5,29}/[a-zA-Z0-9][a-zA-Z0-9_.-]+):([a-zA-Z0-9_.-]+)\b`)
```

Comment:
Those are the restrictions on GCP project names, which is what these are.

Reply:
nit: hardcoding `gcr.io` here means that this cannot be used with e.g. `eu.gcr.io`, and prevents this from being slightly more general purpose.

Given the user provides `--image-regex` as a flag, it's confusing (when I just tried to run this 😄) that no files matched, because my image-regex began with `eu.gcr.io`.

Might not be too important for test-infra, so needn't be a big issue before merging given the experimental nature 😄

Reply:
I can probably accept gcr.io subdomains, too.

Reply:
Anything that looks like a GCR subdomain is now accepted. I am just assuming that eu.gcr.io will expose the same API as gcr.io.